src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = Object.assign({}, state);
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
```

Comment:
It's recommended to use a switch statement for better readability when handling multiple cases based on the same variable